platform/engineering/design-docs/2020-09-24-package-forms-library.md

```diff
@@ -0,0 +1,217 @@
+# Bundling the forms library as a Node package
+
+**Author(s):** Brooks Johnson 
+**Last Updated:** September 24, 2020  
+**Status:** **Draft** | In Review | Approved  
+**Approvers:** _Chris Valarida_ \[ \], _Michael Fleet_ \[ \], ...  
+
+## Overview
+
+### Objective
+
+The goal of this project is to define clear boundaries around the collection of code, markup, and styling that is referred to as "the forms library".
+
+Some non-goals are:
+
+- Bug fixes to the forms library
+- Adding new features to the forms library
+- Refactoring the forms library beyond what is required to group the existing code together in the same package
+- Establishing a separate build & publish pipeline for the forms library (yet)
+
+The intended audience for this document is VSP and VFS frontend engineers.
+
+### Background
+
+The forms library is a collection of React components, CSS, and JS that allows an app team to build a form application for `vets-website` using a combination of JSON schemas.
+
+Right now in `vets-website` we have the following folders containing code for the forms library:
+
+- `src/platform/forms`
+- `src/platform/forms-system`
+
+We also have forms-adjacent code in the form of other dependencies
+
+- `@department-of-veterans-affairs/react-jsonschema-form` (See [Tighter dependencies](#tighter-dependencies))
+- `vets-json-schema`
+
+
+Additionally, the forms library code is entangled with other platform code as well as some app code.
+
+#### Application entanglement
+
+[This definition file](https://github.com/department-of-veterans-affairs/vets-website/blob/720a867817f5b83bd1d713bd51202863b41739b1/src/platform/forms/definitions/nonMilitaryJobs.js#L1) in the forms library imports some app code, yet the file itself is _only_ used in other app code.
+
+There is a [helper file](https://github.com/department-of-veterans-affairs/vets-website/blob/cc4a3172edc242bf60c93aec5b170c734dc57985/src/applications/personalization/dashboard/helpers.jsx) in `src/applications/personalization` which exports a number of consts which are used in the following files within the forms library:
```

Comment:
You already know this, but it's worth capturing here for posterity: I'm already [working on fixing this](https://github.com/department-of-veterans-affairs/vets-website/pull/14468).